openapi_to_fastapi/routes.py

```diff
@@ -0,0 +1,130 @@
+import json
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Callable, Dict, List, Optional, Sequence, Type
+
+import pydantic
+from fastapi import APIRouter
+
+from .model_generator import load_models
+from .models import Operation
+from .parser import parse_openapi_spec
+from .validator import BaseValidator
+from .validator.core import DefaultValidator
+
+
+class EmptyModel(pydantic.BaseModel):
+    pass
+
+
+def make_dummy_route(
+    request_model: Type[pydantic.BaseModel], response_model: Type[pydantic.BaseModel]
+):
+    def _route(request: request_model):  # type: ignore
+        return {}
+
+    return _route
+
+
+@dataclass
+class RouteInfo:
+    description: Optional[str] = None
+    factory: Callable = make_dummy_route
+    name: Optional[str] = None
+    name_factory: Optional[Callable] = None
+    response_description: Optional[str] = None
+    tags: Optional[List[str]] = None
+
+
+@dataclass
+class RoutesMapping:
+    default_get: RouteInfo = RouteInfo()
+    default_post: RouteInfo = RouteInfo()
+    get_map: Optional[Dict[str, RouteInfo]] = None
+    post_map: Optional[Dict[str, RouteInfo]] = None
+
+
+def add_route(
+    api_router: APIRouter,
+    path: str,
+    method: str,
+    operation: Operation,
+    routes: RoutesMapping,
+    models_module,
+):
+
+    resp_model = None
+    if operation.responseModels and operation.responseModels.get(200):
+        resp_model = getattr(models_module, operation.responseModels[200])
+
+    router_method = getattr(api_router, method, None)
+    if not router_method:
+        raise ValueError("Unsupported HTTP method")
```

Comment:
You seem to have custom errors for many things, but then for other things you just use `ValueError`. Is this on purpose?

Reply:
There's no such line anymore, but anyway it was intentional. Custom errors only applied to the parser logic. But there was library misusage